hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -220,8 +223,22 @@ static int enterStopBasedSleep(const hal_sleep_config_t* config, hal_wakeup_sour
     // We need to do this before disabling systick/interrupts, otherwise
     // there is a high chance of a deadlock
     if (config->mode == HAL_SLEEP_MODE_ULTRA_LOW_POWER) {
+        int skipUsart = -1;
```

Comment:
This should not be needed on Gen 2, right? We are entering a low power mode on top of regular WFI/WFE where only EXTI and RTC work anyway.

Reply:
We've introduced a network flag `SystemSleepNetworkFlag::INACTIVE_STANDBY`, which allow the network modem to be on while the device is asleep. When the ULP mode is specified, along with this flag, we should skip suspending the USART that is used by the modem.

Reply:
I understand that, however due to the fact that we are now turning off 1.2V power domain both in STOP and ULP modes, this does not make sense anyway, because the UART is not running. HSE/LSI are also off.

Moreover, we are already correctly putting the UART into a proper state in `cellular_pause()` on Gen 2.

Reply:
That makes perfect sense. 👍

Reply:
So, I propose we rollback this change (at least for now), also since this PR targets Gen 3.